lisa/sut_orchestrator/azure/common.py

```diff
@@ -151,26 +151,55 @@ class AzureVmPurchasePlanSchema:
     publisher: str
 
 
+@dataclass_json
+@dataclass
+class AzureImageSchema:
+    network_data_path: Optional[
+        Union[search_space.SetSpace[schema.NetworkDataPath], schema.NetworkDataPath]
+    ] = field(  # type: ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            items=[
+                schema.NetworkDataPath.Synthetic,
+                schema.NetworkDataPath.Sriov,
+            ],
+        ),
+        metadata=field_metadata(
+            decoder=partial(
+                search_space.decode_set_space_by_type, base_type=schema.NetworkDataPath
+            )
+        ),
+    )
+
+
 @dataclass_json()
 @dataclass
-class AzureVmMarketplaceSchema:
+class AzureVmMarketplaceSchema(AzureImageSchema):
     publisher: str = "Canonical"
     offer: str = "0001-com-ubuntu-server-jammy"
     sku: str = "22_04-lts"
     version: str = "Latest"
+    image_name_keys = ["publisher", "offer", "sku", "version"]
 
     def __hash__(self) -> int:
         return hash(f"{self.publisher}/{self.offer}/{self.sku}/{self.version}")
 
 
 @dataclass_json()
 @dataclass
-class SharedImageGallerySchema:
+class SharedImageGallerySchema(AzureImageSchema):
     subscription_id: str = ""
     resource_group_name: Optional[str] = None
     image_gallery: str = ""
     image_definition: str = ""
     image_version: str = ""
+    image_name_keys = [
```

Comment:
move it out of the class, because it's a data class, all class fields will be regarded as object fields.

Reply:
Yeah, I didn't aware of that. Have modified.